### PR TITLE
ESDT with transfer role

### PR DESCRIPTION
**Pull request by sasurobert**
Implemented ESDT with transfer role

### CODE REVIEW

File: builtInFunctions/esdtMetaData.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ const lengthOfESDTMetadata = 2
 const (
 	// MetadataPaused is the location of paused flag in the esdt global meta data
 	MetadataPaused = 1
+	// MetadataLimitedTransfer is the location of paused flag in the esdt global meta data
+	MetadataLimitedTransfer = 4
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply builtInFunctions/esdtMetaData.go:9]: changed to 2.

File: builtInFunctions/esdtNFTTransfer.go [hunk 2]
```diff
@@ -171,7 +194,7 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
 	if bytes.Equal(dstAddress, vmInput.CallerAddr) {
 		return nil, fmt.Errorf("%w, can not transfer to self", ErrInvalidArguments)
 	}
-	if e.shardCoordinator.ComputeId(dstAddress) == core.MetachainShardId {
+	if e.shardCoordinator.ComputeId(dstAddress) == core.MetachainShardId && !e.flagTransferToMeta.IsSet() {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply builtInFunctions/esdtNFTTransfer.go]: done

File: builtInFunctions/esdtTransfer.go [hunk 3]
```diff
@@ -78,7 +102,7 @@ func (e *esdtTransfer) ProcessBuiltinFunction(
 	if err != nil {
 		return nil, err
 	}
-	if e.shardCoordinator.ComputeId(vmInput.RecipientAddr) == core.MetachainShardId {
+	if e.shardCoordinator.ComputeId(vmInput.RecipientAddr) == core.MetachainShardId && !e.flagTransferToMeta.IsSet() {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply builtInFunctions/esdtTransfer.go]: done

**Review by iulianpascalau — DISMISSED** [review]

File: builtInFunctions/esdtGlobalSettings.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	vmOutput := &vmcommon.VMOutput{ReturnCode: vmcommon.Ok}
+	return vmOutput, nil
+}
+
+func (e *esdtGlobalSettings) toggleSetting(token []byte) error {
+	systemSCAccount, err := e.getSystemAccount()
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply builtInFunctions/esdtGlobalSettings.go:97]: no need. as thorugh base enabled we return IsActive or not for this built in function. So this part is treated well.

File: builtInFunctions/esdtMetaData.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply builtInFunctions/esdtMetaData.go]: ok

File: builtInFunctions/esdtGlobalSettings.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply builtInFunctions/esdtGlobalSettings.go:97]: ok

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**